spec/unit/use_cases/calculate_total_cost_spec.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+describe CalculateTotalCost do
+  let(:file_items_gateway) { spy }
+  let(:use_case) { described_class.new(items_gateway: file_items_gateway) }
+
+  it 'uses the gateway to retrieve the item details' do
+    use_case.execute
+    expect(file_items_gateway).to have_received(:all)
+  end
+
+end
```

Comment:
could we have another test in here to check the overall total? maybe we could create a double for the items gateway